File: cs/sdk/src/Proton.Drive.Sdk/Telemetry/DownloadEvent.cs
using Proton.Sdk.Telemetry;

namespace Proton.Drive.Sdk.Telemetry;

public sealed class DownloadEvent : IMetricEvent
{
    public string Name => "download";

    public required VolumeType VolumeType { get; init; }

    public long DownloadedSize { get; set; }

    public long ClaimedFileSize { get; set; }

    public DownloadError? Error { get; set; }

    public string? OriginalError { get; set; }
}

File: kt/sdk/src/main/kotlin/me/proton/drive/sdk/telemetry/DownloadEvent.kt
package me.proton.drive.sdk.telemetry

data class DownloadEvent(
    val volumeType: VolumeType,
    val claimedFileSize: Long,
    val downloadedSize: Long,
    val error: DownloadError? = null,
    val originalError: String? = null,
)
